Added a fifth form of input as a triad of points.

malibrud opened the pull request:
I find it useful to generate coordinates from a set of three non-colinear points in space.  I have heard this method called a triad.  Basically, two of the points form the x direction, then the y direction  is defined to be in the same direction as the third from the first (note that it does not have to be perpendicular to X, it just defines the X-Y plane and direction of the Z axis.

Mark Brudnak

malibrud committed:
Added a fifth form of input as a triad of points.

gaschler commented on index.html:
```diff
@@ -159,6 +159,27 @@
 <span class="dl2">x</span><input type="number" pattern="\d+(\.\d*)?" id="e0" value="0" oninput="update(4)"/>
 <span class="dl2">y</span><input type="number" pattern="\d+(\.\d*)?" id="e1" value="0" oninput="update(4)"/>
 <span class="dl2">z</span><input type="number" pattern="\d+(\.\d*)?" id="e2" value="0" oninput="update(4)"/>
+<p class="pi" id="ip5">Triad of points, <b>P</b><sub>1</sub>, <b>P</b><sub>2</sub>, <b>P</b><sub>3</sub>, 
```
P should not be bold.

gaschler replied:
Ok, this new type of conversion seems useful.
I only disagree about the notation and naming. Among other things, the book does not say "triad".
Thanks for giving all the details and clarification.
Can we call the title "Coordinate system from three points" and the explanation "Points O, P_X, P_Y define a coordinate system with origin O, P_X on the x-axis, and P_Y on the x-y-plane."?
The only thing that strikes me is the origin is mostly irrelevant to the calculation, maybe it should be left out?

gaschler commented on index.html:
```diff
@@ -296,6 +317,31 @@
+    else if (inputMode ==5)
+    {
+        var m = new THREE.Matrix4();
+        var p1x = document.getElementById("p1x").value;
```
Feel free to introduce a shorter helper function for getElementById...value

malibrud replied:
It was added with getVector().

gaschler reviewed (commented):
Thank you for preparing this new feature.
However, I am not fully convinced that "triad of points" is an often used notation, because it is not mentioned in Wikipedia. Can you point me to documentation?
Especially the term "triad" I find unusual.
How about just saying "Three points such that x-axis is parallel to line P1-P2 and z-axis to line P1-P3"
Also, I cannot think of a case where P1 would be other than "0 0 0".

malibrud commented:
Thank you for the helpful comments.  I encountered the term in setting up multi-body dynamics codes.  In these systems one usually needs to specify a joint position and axis.  For example, a revolute joint (i.e. a hinge) rotates about a fixed line in space.  Most multi-body dynamics code therefore require the modeler to specify this line by specifying a point on the line and a direction.  This is usually done by defining a coordinate system where a specified axis is aligned with the rotation axis.  To specify this line one often measures hard points on the system (i.e. P1, P2, P3).  So one often finds them selves using these points to generate a coordinate system which is rotated and translated.  The so called "triad" is a method to do that where one does not have to concern oneself with orthogonality or normalization.  I have seen this method in commercial dynamics and custom written dynamics codes.

malibrud commented:
Here is a scan of a book [1] which describes the method using a triad (Example 9.1.6).  Here they do not use the term triad, but describe it as a set of points P, Q, R. It may be that I could change the notation to be P, Q, R to be consistent with this author.

![triad_definition](https://user-images.githubusercontent.com/8688159/32394093-59e2b282-c0b2-11e7-8c4b-8630f8fe3af0.png)

[1] Haug, E., "Computer Aided Kinematics and Dynamics of Mechanical Systems," Allyn and Bacon, Boston, 1989.

malibrud committed:
Changed per comments from gaschler.
Removed Triad and replaced with Triple.
Changed points to P, Q, R.
Added a helper function getVector() to simplify extracting the P, Q, R vectors.

malibrud commented:
@gaschler, I posted an updated pull request.

gaschler commented:
@malibrud, thanks for your contribution, I merged it.
